crates/app/src/builder.rs

```diff
@@ -3,32 +3,56 @@
 
 use super::{handle, RepoStore, TagStore, TreeStore};
 
+use drawbridge_store::Filesystem;
+
+use std::error::Error;
+use std::fs::File;
+use std::path::Path;
 use std::sync::Arc;
 
 use axum::handler::Handler;
 use axum::routing::IntoMakeService;
 use axum::{Extension, Router};
+use cap_async_std::fs::Dir;
+use tokio::sync::RwLock;
 
-#[derive(Default)]
-pub struct Builder;
+pub struct Builder<'a> {
+    data_directory: &'a Path,
+}
 
-impl Builder {
-    pub fn new() -> Self {
-        Self
+impl<'a> Builder<'a> {
+    /// Constructs a new [Builder].
+    pub fn new(data_directory: &'a Path) -> Self {
+        Self { data_directory }
     }
 
-    // TODO: Add configuration functionality
-
     /// Builds the application and returns Drawbridge instance as a [tower::MakeService].
-    pub fn build(self) -> IntoMakeService<Router> {
-        let repos: Arc<RepoStore> = Default::default();
-        let tags: Arc<TagStore> = Default::default();
-        let trees: Arc<TreeStore> = Default::default();
-        Router::new()
+    pub fn build(self) -> Result<IntoMakeService<Router>, Box<dyn Error>> {
+        let path = self.data_directory;
+
+        if !path.exists() {
+            eprintln!(
```

Comment:
This should `return`.
Note that you can directly turn a string literal into a `Box<dyn Error>` via `.into()`.
So something like `return Err("<msg>".into())` should work

Reply:
wow I did not know that thanks